dotCMS/src/main/java/com/dotmarketing/servlets/BinaryExporterServlet.java

```diff
@@ -192,123 +195,135 @@ public void doGet(HttpServletRequest req, HttpServletResponse resp) throws Servl
-                        resp.sendError(404);
-                        return;
-                    }
-                }
+			// If the user is NOT logged in the backend then we cannot show content that is NOT live.
+			// Temporal files should be allowed any time
+			if(!isTempBinaryImage && !WebAPILocator.getUserWebAPI().isLoggedToBackend(req)) {
+			    if (!APILocator.getVersionableAPI().hasLiveVersion(content) && respectFrontendRoles) {
+			        Logger.debug(this, "Content " + fieldVarName + " is not published, with inode: "
+			                + content.getInode());
+			        resp.sendError(HttpServletResponse.SC_NOT_FOUND);
+			        return;
+			    }
+			}
 
-				//Find the contentlet content type
-				ContentType type = APILocator.getContentTypeAPI(APILocator.systemUser()).find((content.getContentTypeId()));
-				//And the file asset field
-				com.dotcms.contenttype.model.field.Field field;
+			//Find the contentlet content type
+			ContentType type = APILocator.getContentTypeAPI(APILocator.systemUser()).find((content.getContentTypeId()));
+			
+			if(type.baseType() != BaseContentType.FILEASSET){
+                Logger.debug(this,"Content is not a File-Type Content Type.");
+                resp.sendError(HttpServletResponse.SC_NOT_FOUND);
+                return;
+            }
+			
+			//And the file asset field
+			com.dotcms.contenttype.model.field.Field field;
 
-				try {
-					field = APILocator.getContentTypeFieldAPI().byContentTypeAndVar(type, fieldVarName);
-				} catch (NotFoundInDbException e) {
-					Logger.debug(this,"Field " + fieldVarName + " does not exist within structure " + type.variable());
-					resp.sendError(404);
-					return;
-				}
+			try {
+			    field = APILocator.getContentTypeFieldAPI().byContentTypeAndVar(type, fieldVarName);
+			} catch (NotFoundInDbException e) {
+			    Logger.debug(this,"Field " + fieldVarName + " does not exist within structure " + type.variable());
+			    resp.sendError(HttpServletResponse.SC_NOT_FOUND);
+			    return;
+			}
 
-				if(isTempBinaryImage)
-					inputFile = contentAPI.getBinaryFile(content.getInode(), field.variable(), APILocator.getUserAPI().getSystemUser());
-				else
-					inputFile = contentAPI.getBinaryFile(content.getInode(), field.variable(), user);
-				if(inputFile == null){
-					Logger.debug(this,"binary file '" + fieldVarName + "' does not exist for inode " + content.getInode());
-					resp.sendError(404);
-					return;
-				}
-				downloadName = inputFile.getName();
+			if(isTempBinaryImage) {
+			    inputFile = contentAPI.getBinaryFile(content.getInode(), field.variable(), APILocator.systemUser());
+			} else {
+			    inputFile = contentAPI.getBinaryFile(content.getInode(), field.variable(), user);
+			}
+			if(inputFile == null){
+			    Logger.debug(this,"binary file '" + fieldVarName + "' does not exist for inode " + content.getInode());
+			    resp.sendError(HttpServletResponse.SC_NOT_FOUND);

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Add a "try/catch" block for "sendError". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1989)